realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -830,34 +865,46 @@ void BaseRealSenseNode::fillMessageImage(
     const stream_index_pair& stream,
     unsigned int width,
     unsigned int height,
-    unsigned int bpp,
+    const rs2_format& stream_format,
     const rclcpp::Time& t,
     sensor_msgs::msg::Image* img_msg_ptr)
 {
+    if(rs_format_to_ros_format.find(stream_format) == rs_format_to_ros_format.end())
+    {
+        ROS_ERROR_STREAM("Format " << rs2_format_to_string(stream_format) << " is not supported in ROS2 image messages");
+    }
     // Convert the CV::Mat into a ROS image message (1 copy is done here)
-    cv_bridge::CvImage(std_msgs::msg::Header(), _encoding.at(bpp), cv_matrix_image).toImageMsg(*img_msg_ptr);
+    cv_bridge::CvImage(std_msgs::msg::Header(), rs_format_to_ros_format[stream_format], cv_matrix_image).toImageMsg(*img_msg_ptr);
```

Comment:
For unsupported format this will create a new entry, probably initialized with RS2_FORMAT_ANY. Is this OK?

Reply:
Unsupported formats will be stopped several steps before, and will switch the stream format to the default one (for example, for color, it will be RGB as default)

Reply:
I saw that you check above, but as I understand `ROS_ERROR_STREAM` does not throw (otherwise [this](https://github.com/IntelRealSense/realsense-ros/blob/ros2-development/realsense2_camera/src/base_realsense_node.cpp#L163) won't work) and you don't return.

Reply:
Fixed functions to return status, and checked status after
(did this for two functions: when filling CV matrix image, and when filling ROS image msg)